packages/discovery/src/discovery/provider/DiscoveryProvider.ts

```diff
@@ -56,13 +59,56 @@ export class DiscoveryProvider {
     return Bytes.fromHex(result)
   }
 
-  async getLogs(
+  public async getLogs(
     address: EthereumAddress,
     topics: (string | string[])[],
     fromBlock: number,
     toBlock: number,
   ): Promise<providers.Log[]> {
-    return this.provider.getLogs({
+    if (fromBlock > toBlock) {
+      throw new Error(
+        `fromBlock (${fromBlock}) can't be bigger than toBlock (${toBlock})`,
+      )
+    }
+
+    if (this.getLogsMaxRange === undefined) {
+      return await this.getLogsBatch(address, topics, fromBlock, toBlock)
+    }
+
+    // To support efficient caching, we divide the requested blocks range into
+    // sequential boundaries of `maxRange` size, e.g [0,10k-1], [10k, 20k-1], ...
+    // Otherwise ranges would depend on `fromBlock` and even small change to it
+    // would make the previous cache useless.
```

Comment:
What would happen to the cache in the following scenario?

- We start using QuickNode with `maxRange = 10k`
- We switch to QuickNode2 that has `maxRange = 5k`

Is the cache useless and has to be regenerated?

Reply:
Yes, changing maxRange would make previous cache useless. I don't see a simple solution for that, except for a custom cache for events (caching events separately with block numbers and grouping on request etc.) but that is not easy to do and can be done as a different story.